bagel/tests/test_utility.py

```diff
@@ -17,6 +17,46 @@ def get_test_context():
     return putil.generate_context()
 
 
+@pytest.fixture
+def get_values_by_key():
+    """
+    Get values of all instances of a specified key in a dictionary. Will also look inside lists of dictionaries and nested dictionaries.
+    """
+
+    def _find_by_key(data, target):
+        if isinstance(data, dict):
+            for key, value in data.items():
+                if isinstance(value, (dict, list)):
+                    yield from _find_by_key(value, target)
+                elif key == target:
+                    yield value
+        elif isinstance(data, list):
+            for item in data:
+                yield from _find_by_key(item, target)
+
+    return _find_by_key
+
+
+def test_all_used_namespaces_have_urls(
+    get_test_context, get_values_by_key, load_test_json, test_data_upload_path
+):
+    """Test that all namespace prefixes used in a comprehensive data dictionary have a corresponding URL in the @context."""
+    data_dict = load_test_json(
+        test_data_upload_path / "example_synthetic.json"
+    )
+    prefixes = list(get_values_by_key(data_dict, "TermURL"))
+
+    for idx, _ in enumerate(prefixes):
+        prefixes[idx] = prefixes[idx].split(":")[0]
```

Comment:
:cherries: but I'd find this cleaner

```suggestion
prefixes = list(map(lambda term: term.split(":")[0], 
                                  get_values_by_key(data_dict, "TermURL")))
```